beacon-chain/p2p/service.go

```diff
@@ -448,3 +457,9 @@ func (s *Service) connectToBootnodes() error {
 	s.connectWithAllPeers(multiAddresses)
 	return nil
 }
+
+// Returns true if the service is aware of the genesis time and genesis validator root. This is
+// required for discovery and pubsub validation.
+func (s *Service) isInitialized() bool {
+	return !s.genesisTime.IsZero() && len(s.genesisValidatorsRoot) == 32
```

Comment:
I dont think we need to add this helper in. P2P service never starts until state is initialized.

Reply:
Discussed offline, this isn't true. I was able getting e2e failures where the sync service was trying to subscribe to topics before the p2p service had fully initialized. Adding this check is an added safety against the possible race condition